src/data/nat/basic.lean

```diff
@@ -308,6 +311,32 @@ begin
   exact mul_le_mul_of_nonneg_left (nat.succ_le_of_lt h) dec_trivial,
 end
 
+theorem odd_lt_even_of_lt : Π {n m}, m < n → 2 * m + 1 < 2 * n
+| 0 m h := absurd h $ not_lt_zero _
+| (n+1) m h := 
+begin 
+  cases eq_or_lt_of_le (le_of_lt_succ h) with heq hlt, 
+  { simp [heq, mul_add], exact dec_trivial },
+  { apply lt_trans (odd_lt_even_of_lt hlt), 
+    simp [mul_add], 
+    exact dec_trivial }
+end
+
+theorem even_ne_odd {n m} : 2 * n ≠ 2 * m + 1 :=
```

Comment:
Shorter proof 
`mt (congr_arg (%2)) (by rw [add_comm, add_mul_mod_self_left, mul_mod_right]; exact dec_trivial)`

Reply:
Looks great! Updated.

Reply:
What is this `%2`?

Reply:
It’s just `(λ x, x % 2)`.

Reply:
It's longer, but would `two_mul_ne_two_mul_add` be a better name for this?

Reply:
Both look reasonable to me, but `two_mul_ne_two_mul_add` should be more conventional. In case of the latter, we might as well say `two_mul_ne_two_mul_add_one`?

Reply:
Or `bit0_ne_bit1`?